mobile/lib/services/sync.service.dart

```diff
@@ -220,8 +227,31 @@ class SyncService {
     return null;
   }
 
+  Future<void> _moveToTrashMatchedAssets(Iterable<String> idsToDelete) async {
+    final List<Asset> assets =
+        await _assetRepository.getAllByRemoteId(idsToDelete);
+
+    if (assets.isNotEmpty) {
+      final localAssets = await _assetRepository.getAllLocal();
+
+      // Find the local assets that match the remote assets by fileName
+      final matchedAssets = localAssets
+          .where(
+            (localAsset) => assets.any(
+              (remoteAsset) =>
+                  remoteAsset.fileName.contains(localAsset.fileName),
+            ),
+          )
+          .toList();
+
+      for (var asset in matchedAssets) {
+        FileTrashManager.moveToTrash(asset.fileName);
```

Comment:
You should create a file repository and move all methods from the `FileTrashManager` library in there. It helps with mocking the library for testing